hedera-node/hedera-app/src/test/java/com/hedera/node/app/grpc/impl/usage/GrpcUsageTrackerTest.java

```diff
@@ -0,0 +1,226 @@
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.util.List;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.concurrent.atomic.LongAdder;
+import org.apache.logging.log4j.LogManager;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+class GrpcUsageTrackerTest {
+
+    private static final Key<String> userAgentHeaderKey = Key.of("X-User-Agent", Metadata.ASCII_STRING_MARSHALLER);
+
+    static LogCaptor accessLogCaptor = new LogCaptor(LogManager.getLogger("grpc-access-log"));
+
+    static final VarHandle usageBucketRefHandle;
+
+    static {
+        try {
+            usageBucketRefHandle = MethodHandles.privateLookupIn(GrpcUsageTracker.class, MethodHandles.lookup())
+                    .findVarHandle(GrpcUsageTracker.class, "bucketRef", AtomicReference.class);
+        } catch (final Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @AfterEach
+    void afterEach() {
+        accessLogCaptor.stopCapture();
+        accessLogCaptor = new LogCaptor(LogManager.getLogger("grpc-access-log"));
+    }
+
+    @AfterAll
+    static void afterAll() {
+        accessLogCaptor.stopCapture();
+    }
+
+    @Test
+    void testInterceptDisabled() {
+        final Clock clock = Clock.fixed(Instant.parse("2025-04-03T15:32:32.426457Z"), ZoneOffset.UTC);
+        final GrpcUsageTrackerConfig config = new GrpcUsageTrackerConfig(false, 15, 100);
+        final ConfigProvider configProvider = mock(ConfigProvider.class);
+        final VersionedConfiguration configuration = mock(VersionedConfiguration.class);
+        final ServerCall<String, String> serverCall = mock(ServerCall.class);
+        final ServerCallHandler<String, String> handler = mock(ServerCallHandler.class);
+        final Metadata metadata = new Metadata();
+        metadata.put(userAgentHeaderKey, "hiero-sdk-java/2.3.1");
+        final MethodDescriptor<String, String> descriptor = newDescriptor("proto.MyService/commit");
+
+        when(configProvider.getConfiguration()).thenReturn(configuration);
+        when(configuration.getConfigData(GrpcUsageTrackerConfig.class)).thenReturn(config);
+        when(serverCall.getMethodDescriptor()).thenReturn(descriptor);
+
+        final GrpcUsageTracker usageTracker = new GrpcUsageTracker(configProvider, clock);
+
+        assertDoesNotThrow(() -> usageTracker.interceptCall(serverCall, metadata, handler));
+
+        // get the usage bucket... there should be no data captured in it since usage tracking is disabled
+        final AtomicReference<UsageBucket> bucketRef =
+                (AtomicReference<UsageBucket>) usageBucketRefHandle.get(usageTracker);
+        assertNotNull(bucketRef);
```

Comment:
I don't find `assertNotNull` to be helpful unless an accompanying message explaining _why_ it might be null is included. Otherwise it adds extra lines to the test without any benefit. `assertTrue(usageBucket.usageData()...)` will fail with the same NPE _and_ implies `usageBucket` should not be null.